java/kotlin-extractor/src/main/kotlin/KotlinUsesExtractor.kt

```diff
@@ -1277,10 +1277,7 @@ open class KotlinUsesExtractor(
                         javaClass.declarations.findSubType<IrFunction> { decl ->
                             decl.name.asString() == jvmName &&
                             decl.valueParameters.size == f.valueParameters.size &&
-                            // Note matching by classifier not the whole type so that generic arguments are allowed to differ,
-                            // as they always will for method type parameters occurring in parameter types (e.g. <T> toArray(T[] array)
-                            // Differing only by nullability would also be insignificant if it came up.
-                            decl.valueParameters.zip(f.valueParameters).all { p -> p.first.type.classifierOrNull == p.second.type.classifierOrNull }
+                            decl.valueParameters.zip(f.valueParameters).all { p -> erase(p.first.type) == erase(p.second.type) }
```

Comment:
@smowton I realized that I should have kept the `.classifierOrNull`, because we do want to ignore nullability. I'll include this in a followup PR. Possibly in https://github.com/github/codeql/pull/10646.